### PR TITLE
[UT]: Fix `test_propagate_nan`

### PR DESCRIPTION
**Pull request by quintinwang5**
Fix test_propagate_nan

### CODE REVIEW

File: lib/Conversion/TritonGPUToLLVM/ElementwiseOpToLLVM.cpp [hunk 1]
```diff
@@ -2294,15 +2294,21 @@ struct MinMaxFOpConversion
   using Adaptor = typename Base::OpAdaptor;
 
   static_assert(std::is_same<OpTy, arith::MinimumFOp>::value ||
-                    std::is_same<OpTy, arith::MaximumFOp>::value,
-                "OpTy must be arith::MinimumFOp or arith::MaximumFOp");
+                    std::is_same<OpTy, arith::MaximumFOp>::value ||
+                    std::is_same<OpTy, arith::MinNumFOp>::value ||
+                    std::is_same<OpTy, arith::MaxNumFOp>::value,
+                "OpTy must be arith::MinimumFOp/arith::MaximumFOp or "
+                "LLVM::MinNumFOp/LLVM::MaxNumFOp");
```
Comment (whitneywhtsang) [inline lib/Conversion/TritonGPUToLLVM/ElementwiseOpToLLVM.cpp:2301]:
```suggestion
                "arith::MinNumFOp/arith::MaxNumFOp");
```

File: lib/Conversion/TritonGPUToLLVM/ElementwiseOpToLLVM.cpp
Comment (whitneywhtsang) [inline lib/Conversion/TritonGPUToLLVM/ElementwiseOpToLLVM.cpp:1]: We usually guide our changes under `target == Target::GENX`.

File: lib/Conversion/TritonGPUToLLVM/ElementwiseOpToLLVM.cpp [hunk 2]
```diff
@@ -2627,8 +2634,19 @@ void mlir::triton::populateElementwiseOpToLLVMPatterns(
                                       benefit);
   patterns.add<ClampFOpConversion>(typeConverter, axisInfoAnalysis,
                                    computeCapability, target);
+  // TODO(FIXME): spirv's OpenCL extension (fmin/fmax) does not support
```
Comment (whitneywhtsang) [inline lib/Conversion/TritonGPUToLLVM/ElementwiseOpToLLVM.cpp:2637]: Which patterns creates the spirv's OpenCL extension (fmin/fmax)?

Comment (quintinwang5) [reply lib/Conversion/TritonGPUToLLVM/ElementwiseOpToLLVM.cpp:2637]: It should be the llvm-spirv translator.

Comment (whitneywhtsang) [reply lib/Conversion/TritonGPUToLLVM/ElementwiseOpToLLVM.cpp:2637]: If not set the high benefit, then the `arith` operators would be lowered by `ArithToLLVMConversionPatterns`.

File: lib/Conversion/TritonGPUToLLVM/ElementwiseOpToLLVM.cpp [hunk 3]
```diff
@@ -2627,8 +2634,19 @@ void mlir::triton::populateElementwiseOpToLLVMPatterns(
...
   patterns.add<MinMaxFOpConversion<arith::MaximumFOp>>(
-      typeConverter, axisInfoAnalysis, computeCapability, target);
+      typeConverter, axisInfoAnalysis, computeCapability, target,
+      benefit = 65534);
+  patterns.add<MinMaxFOpConversion<arith::MinNumFOp>>(
+      typeConverter, axisInfoAnalysis, computeCapability, target,
+      benefit = 65534);
+  patterns.add<MinMaxFOpConversion<arith::MaxNumFOp>>(
+      typeConverter, axisInfoAnalysis, computeCapability, target,
+      benefit = 65534);
```
Comment (whitneywhtsang) [inline lib/Conversion/TritonGPUToLLVM/ElementwiseOpToLLVM.cpp:2651]: From the modified pattern, 
`arith::MinNumFOp` is lowered to `LLVM::MinNumFOp`
`arith::MaxNumFOp` is lowered to `LLVM::MaxNumFOp`
which is the same as (at line 2561)
```
  POPULATE_BINARY_OP(
      arith::MinNumFOp,
      LLVM::MinNumOp) // fmin (return non-NaN if either op is non-NaN)
  POPULATE_BINARY_OP(
      arith::MaxNumFOp,
      LLVM::MaxNumOp) // fmax (return non-NaN if either op is non-NaN)
```
Do we really need these 2 patterns?
Verified that `test_propagate_nan` still pass without these 2 patterns.

Comment (quintinwang5) [reply lib/Conversion/TritonGPUToLLVM/ElementwiseOpToLLVM.cpp:2651]: Actually not. They have the same behavior.

Comment (quintinwang5) [reply lib/Conversion/TritonGPUToLLVM/ElementwiseOpToLLVM.cpp]: Just update the benefit for `GENX` target is enough. So the pattern was definitely not work previously.

**Review by whitneywhtsang — APPROVED** [review]

**Review by etiotto — COMMENTED** [review]
Cool! LGTM

**Pull request merged**